aif360/detectors/ot_detector.py

```diff
@@ -0,0 +1,257 @@
+        matrix_distance = cost_matrix
+    else:
+        matrix_distance = np.array([abs(i - required_distribution) for i in initial_distribution], dtype=float)
+    return initial_distribution, required_distribution, matrix_distance
+
+def _evaluate(
+        ground_truth: pd.Series,
+        classifier: pd.Series,
+        sensitive_attribute: pd.Series=None,
+        data: pd.DataFrame=None,
+        num_iters=1e5,
+        **kwargs):
+    """If the given golden_standart and classifier are distributions, it returns the Wasserstein distance between them, 
+    otherwise it extract all neccessary information from data, makes logistic regression and 
+    compute optimal transport for all possible options for the given classifier.
+
+    Args:
+        ground_truth (pd.Series, str): ground truth (correct) target value
+        classifier (pd.Series): estimated target values
+        sensitive_attribute (pd.Series, str): pandas series of sensitive attribute values
+        data (dataframe): the dataset (containing the features) the model was trained on; \
+              None if `ground_truth`, `classifier` and `sensitive_attribute` are `pd.Series`
+        num_iters (int, optional): number of iterations (random restarts). Should be positive.
+
+    Returns:
+        ot.emd2 (float, dict): Earth mover's distance or dictionary of optimal transports for each of option of classifier
+    """
+
+    # Calculate just the EMD between ground_truth and classifier
+    if sensitive_attribute is None:
+        initial_distribution, required_distribution, matrix_distance = _transform(ground_truth, classifier, data, kwargs.get("cost_matrix"))
+        return ot.emd2(a=initial_distribution, b=required_distribution, M=matrix_distance, numItermax=num_iters)
+    
+    if not ground_truth.nunique() == 2:
+        raise ValueError(f"Expected to have exactly 2 target values, got {len(set(data[ground_truth]))}.")
+    
+    # Calculate EMD between ground truth distribution and distribution of each group
+    emds = {}
+    for sa_val in sorted(sensitive_attribute.unique()):
+        initial_distribution = ground_truth[sensitive_attribute == sa_val]
+        required_distribution = classifier[sensitive_attribute == sa_val]
+        initial_distribution, required_distribution, matrix_distance = _transform(initial_distribution, required_distribution, data, kwargs.get("cost_matrix"))
+        emds[sa_val] = ot.emd2(a=initial_distribution, b=required_distribution, M=matrix_distance, numItermax=num_iters)
+
+    return emds
+    
+
+# Function called by the user
+def ot_bias_scan(
+    ground_truth: Union[pd.Series, str],
+    classifier: Union[pd.Series, str],
+    sensitive_attribute: Union[pd.Series, str] = None,
+    data: pd.DataFrame = None,
+    favorable_value: Union[str, float] = None,
+    overpredicted: bool = True,
+    scoring: str = "Optimal Transport",
+    num_iters: int = 1e5,
+    penalty: float = 1e-17,
+    mode: str = "binary",
+    **kwargs,
```

Comment:
`overpredicted` is indeed unnecessary - will remove.
We do have plans for future `penalty` and `scoring` options (The multiplier associated with the ergodic term in the Sinkhorn algorithm, and Wasserstein-1 and Wasserstein-2 options, respectively)
We suggest to keep `kwargs` as a way to incorporate the less used parameters that may arise in the future. This is the case with `cost_matrix`, as it is rarely necessary, but can still be useful in some cases.

Reply:
I see so `cost_matrix` is specific to this `scoring` option? If other scoring options don't need this then it's fine to keep it under `kwargs` but if it's universal for any scoring function, just rare, it should be named. The only reason to have `kwargs` is to pass `scoring`-specific args which we can't know ahead of time since there are multiple possible scorers.